jans-core/document-store/src/main/java/io/jans/service/document/store/provider/DBDocumentStoreProvider.java

```diff
@@ -0,0 +1,213 @@
+				documentService.addOxDocument(oxDocument);
+				return true;
+			} finally {
+			}
+		} catch (Exception ex) {
+			log.error("Failed to write document to file '{}'", name, ex);
+		}
+
+		return false;
+	}
+
+	@Override
+	public boolean saveDocumentStream(String name, InputStream documentStream, List <String> moduleList) {
+		
+		OxDocument oxDocument = new OxDocument();
+		oxDocument.setDisplayName(name);
+		
+		 try {
+			String documentContent = Base64.getEncoder().encodeToString(IOUtils.toByteArray(documentStream));
+			oxDocument.setDocument(documentContent);
+			String inum = documentService.generateInumForNewOxDocument();
+			oxDocument.setInum(inum);	
+			String dn = "inum="+ oxDocument.getInum() +",ou=document,o=gluu";
+			oxDocument.setDn(dn);
+			oxDocument.setDescription(name);
+			oxDocument.setOxEnabled("true");
+			oxDocument.setOxModuleProperty(moduleList);
+			documentService.addOxDocument(oxDocument);
+			return true;
+		} catch (IOException e) {
+			log.error("Failed to write document from stream to file '{}'", name, e);
+		}catch (Exception e) {
+			log.error("Failed to write document from stream to file '{}'", name, e);
+		}	
+
+		return false;
+	}
+
+
+	@Override
+	public String readDocument(String name, Charset charset) {
+		log.debug("Read document: '{}'", name);		
+		OxDocument oxDocument;
+		try {
+			oxDocument = documentService.getOxDocumentByDisplayName(name);
+			if(oxDocument != null) {
+				return oxDocument.getDocument();
+			}
+		} catch (Exception e) {
+			log.error("Failed to read document as stream from file '{}'", name, e);
+		}
+		return null;		
+	}
+
+	@Override
+	public InputStream readDocumentAsStream(String name) {
+		log.debug("Read document as stream: '{}'", name);
+		String filecontecnt = readDocument(name, null);
+		if (filecontecnt == null) {
+			log.error("Document file '{}' isn't exist", name);
+			return null;
+		}
+
+		InputStream InputStream = new ByteArrayInputStream(Base64.getDecoder().decode(filecontecnt));
+		return InputStream;
+	}
+
+	@Override
+	public boolean renameDocument(String currentDisplayName, String destinationDisplayName) {
+		log.debug("Rename document: '{}' -> '{}'", currentDisplayName, destinationDisplayName);
+		OxDocument oxDocument;
+		try {
+			oxDocument = documentService.getOxDocumentByDisplayName(currentDisplayName);
+			if (oxDocument == null) {
+				log.error("Document doesn't Exist with the name  '{}'", currentDisplayName);
+				return false;
+			}
+			oxDocument.setDisplayName(destinationDisplayName);
+			documentService.updateOxDocument(oxDocument);
+			OxDocument oxDocumentDestination = documentService.getOxDocumentByDisplayName(destinationDisplayName);
+			if(oxDocumentDestination == null) {
+				log.error("Failed to rename to destination file '{}'", destinationDisplayName);
+				return false;
+			}
+		} catch (Exception e) {
+			log.error("Failed to rename to destination file '{}'", destinationDisplayName);
```

Comment:
Something is wrong here. In case of exception, we got further in line 187 and return `true` even if it's not renamed.
Also we should log exception for easier troubleshooting.